homeassistant/components/stream/core.py

```diff
@@ -110,41 +120,71 @@ def idle(self) -> bool:
         return self._idle_timer.idle
 
     @property
-    def segments(self) -> list[int]:
+    def sequences(self) -> list[int]:
         """Return current sequence from segments."""
         return [s.sequence for s in self._segments]
 
+    @property
+    def last_segment(self) -> Segment:
+        """Return the last segment without iterating."""
+        if self._segments:
+            return self._segments[-1]
+        return None
+
+    @property
+    def last_sequence(self) -> int:
+        """Return the last sequence number without iterating."""
+        if self._segments:
+            return self._segments[-1].sequence
+        return -1
+
     @property
     def target_duration(self) -> int:
         """Return the max duration of any given segment in seconds."""
-        segment_length = len(self._segments)
-        if not segment_length:
-            return 1
-        durations = [s.duration for s in self._segments]
-        return round(max(durations)) or 1
+        durations = [s.duration for s in self._segments if s.duration]
+        if not durations:
+            return MIN_SEGMENT_DURATION
+        return round(max(durations)) or MIN_SEGMENT_DURATION
 
     def get_segment(self, sequence: int = None) -> Any:
         """Retrieve a specific segment, or the whole list."""
         self._idle_timer.awake()
 
-        if not sequence:
+        if sequence is None:
             return self._segments
 
         for segment in self._segments:
             if segment.sequence == sequence:
                 return segment
         return None
 
-    async def recv(self) -> Segment:
+    async def part_recv(self, timeout=None) -> bool:
+        """Wait for an event signalling the latest part segment."""
+        try:
+            async with async_timeout.timeout(timeout):
+                await self._part_event.wait()
+        except asyncio.TimeoutError:
+            return False
+        return True
+
+    def part_put(self) -> None:
+        """Set event signalling the latest part segment."""
+        self._part_event.set()
+        self._part_event.clear()
+
+    async def recv(self, timeout=None) -> Segment:
```

Comment:
Used the new last_sequence property to avoid iterating. The timeout has been added in order to support blocking playlist requests.